src/content/app/genome-browser/components/track-panel/components/track-panel-list/track-panel-items/TrackPanelGene.tsx

```diff
@@ -63,67 +66,74 @@ const TrackPanelGene = (props: TrackPanelGeneProps) => {
     getBrowserTrackState(state, {
       genomeId,
       objectId: focusObjectId,
-      tracksGroup: 'objectTracks',
-      categoryName: 'main',
-      trackId: GENE_TRACK_ID
+      tracksGroup: 'objectTracks'
     })
   );
-  const { toggleTrack, genomeBrowser } = useGenomeBrowser();
-
+  const visibleTranscriptIds = useAppSelector(getVisibleTranscriptIds);
+  const { toggleTrack, updateFocusGeneTranscripts, genomeBrowser } =
+    useGenomeBrowser();
   const dispatch = useAppDispatch();
 
+  let allTranscriptsInGene = currentData?.gene.transcripts ?? [];
+  let sortedTranscripts: TrackPanelTranscriptType[] | undefined;
+
   useEffect(() => {
     toggleTrack({ trackId: GENE_TRACK_ID, status: trackStatus });
-  }, [genomeBrowser]);
 
-  if (!currentData) {
-    return null;
-  }
+    const subscription = genomeBrowser?.subscribe(
+      IncomingActionType.VISIBLE_TRANSCRIPTS,
+      (action: ReportVisibleTranscriptsAction) => {
+        setVisibleTranscriptIds(action.payload.transcript_ids);
+      }
+    );
 
-  const toggleExpand = () => {
-    setIsCollapsed(!isCollapsed);
-  };
+    return () => subscription?.unsubscribe();
+  }, [genomeBrowser, genomeId, focusObjectId]);
```

Comment:
Only the `toggleTrack` in this useEffect depends on `genomeId` and `focusObjectId` (AND, on the trackStatus, I would have thought). The subscription to the genome browser, on the other hand, depends only on the presence of the genome browser. These should be two different useEffects.

Reply:
✅